pyproject.toml

```diff
@@ -1,3 +1,171 @@
 [build-system]
-requires = ["setuptools>=42.0.0", "wheel>=0.34.2", "cython>=0.25"]
 build-backend = "setuptools.build_meta"
+requires = [
+    "cython>=0.25",
+    "setuptools>=63",
+]
+
+[project]
+name = "pytential"
+version = "2020.2"
+description = "Evaluate layer and volume potentials accurately. Solve integral equations."
+readme = "README.rst"
+authors = [
+    { name = "Andreas Kloeckner", email = "inform@tiker.net" },
+]
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Other Audience",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Visualization",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Utilities",
+]
+dependencies = [
+    "arraycontext>=2021.1",
+    "boxtree>=2021.1",
+    "immutables>=0.19",
+    "loopy>=2024.1",
+    "meshmode>=2021.2",
+    "modepy>=2021.1",
+    "pymbolic>=2022.2",
+    "pyopencl>=2022.1",
+    "pytools>=2022.1",
+    "scipy>=1.2",
+    "sumpy>=2022.1",
+]
+
+[project.optional-dependencies]
+doc = [
+    "furo",
+    "sphinx>=4",
+    "sphinx-copybutton",
+]
+fmmlib = [
+    "pyfmmlib>=2023.1",
+]
+test = [
+    "mypy",
+    "pylint",
+    "pytest",
+    "ruff",
+]
+
+[project.urls]
+Documentation = "https://documen.tician.de/pytential"
+Homepage = "https://mathema.tician.de/software/pytential"
+Repository = "https://github.com/inducer/pytential"
+
+[tool.setuptools.packages.find]
+include = [
+    "pytential*",
+]
+
+[tool.ruff]
+preview = true
+
+[tool.ruff.lint]
+extend-select = [
+    "B",    # flake8-bugbear
+    "C",    # flake8-comprehensions
+    # "D",    # pydocstyle
+    "E",    # pycodestyle
+    "F",    # pyflakes
+    "G",    # flake8-logging-format
+    # "I",    # flake8-isort
```

Comment:
The isort fixes are not added because that would likely cause a lot of conflicts in the opened PRs :(